newapi/tests/integ/test_prioritization_nodb.py

```diff
@@ -0,0 +1,136 @@
+"""
+Integration test for URL prioritization with mocked database
+Uses data from tests/integ/data/
+Lint using Black.
+Test using:
+    pytest-3 -s --show-capture=no ooniapi/tests/integ/test_prioritization_nodb.py
+"""
+
+import json
+from pathlib import Path
+
+import pytest
+from mock import MagicMock
+
+# Extract database responses with:
+# COPY (SELECT json_agg(t) FROM (...) t) TO '/tmp/<name>.json';
+
+queries = {
+    (
+        "SELECT category_code, priority, url FROM citizenlab WHERE cc = 'ZZ' ORDER BY priority DESC"
+    ): Path("citizenlab.json"),
+    (
+        "SELECT category_code, domain, url, cc, COALESCE(msmt_cnt, 0)::float AS msmt_cnt FROM ( SELECT domain, url, cc, category_code FROM citizenlab WHERE citizenlab.cc = :cc_low OR citizenlab.cc = :cc OR citizenlab.cc = 'ZZ' ) AS citiz LEFT OUTER JOIN ( SELECT input, msmt_cnt FROM counters_test_list WHERE probe_cc = :cc ) AS cnt ON (citiz.url = cnt.input)"
+    ): Path("citizenlab_counters_us.json"),
+    (
+        "SELECT category_code, cc, domain, url, priority FROM url_priorities WHERE cc = :cc OR cc = '*'"
+    ): Path("url_priorities_us.json"),
+}
+
+
```

Comment:
Since these tests are based on deterministic database dumps, perhaps we can also add a check which verifies that the sorting of the returned URLs matches the expected result (i.e. the priorities are in fact being respected)?